### ISSUE TITLE
Implement support for parameters within messages

### ISSUE THREAD

**Issue by ryancox** [balerter/balerter#54]
It would be quite helpful for a number of use cases if [messages ](https://github.com/balerter/balerter/blob/master/internal/message/message.go) could contain an optional set of key value pairs. This would allow for scripts to extract values and dispatch them for later use. For example, webhooks could use these values in the body of the Post payload. Or slack/discord messages could embed them to add more context to alerts. 

Elastalert supports something similar to this whereby an alert can extract field values and then use them to add context to alerts. 

[Example](https://github.com/jertel/elastalert2/blob/master/examples/rules/ssh-repeat-offender.yaml#L30-L38)

```
include:
  - match_body.host.hostname
  - match_body.user.name
  - match_body.source.ip

alert_subject: "SSH abuse (repeat offender) on <{}> | <{}|Show Dashboard>"
alert_subject_args:
  - match_body.host.hostname
  - kibana_link
```

**Comment by negasus** [comment]
I already had this feature. But I removed it for the following reason: some channels cannot receive additional fields. This led to some inconsistency. 

By the way, I think, we can return this feature back

**Comment by ryancox** [comment]
Yeah that makes sense. These channels could ignore the parameters or perhaps a 'ParameterizedMessage' type could be introduced that contained parameters to make things clearer. But that could add complexity to implementations.

**Comment by negasus** [comment]
I am already doing a basic implementation to support fields. First, I'll make support for Telegram and Slack channels for the next release. Then other channels

**Comment by negasus** [comment]
Implemented in v0.8.4

Not support for channels: `notify`, `twilio`
